### PR TITLE
drivers:platform:mbed: I2C platform driver implementation

### PR DESCRIPTION
**Pull request by mphalke**
Initial implementation of the Mbed I2C platform drivers.

Signed-off-by: mahphalke <Mahesh.Phalke@analog.com>

### CODE REVIEW

File: drivers/platform/mbed/mbed_i2c.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+	i2c = (I2C *)(((mbed_i2c_desc *)(desc->extra))->i2c_port);
+
+	/**
+	    The MBED I2C API is reversed for parameter 4
+	    Instead of stop_bit - it has
+	    @param repeated   - Repeated start, true - don't send stop at end default value is false.
+	    Inverting here to keep the no-OS/platform_drivers API
+	 */
+	if (!(i2c->write(desc->slave_address, (char *)data, bytes_number, !stop_bit))) {
```
Comment (amiclaus) [inline drivers/platform/mbed/mbed_i2c.cpp:161]: nit: no need for brackets in this `if ... else` statement. actually you can also get rid of `else`:
```
if ((i2c->write(desc->slave_address, (char *)data, bytes_number, !stop_bit)))
     return -EINVAL;

return SUCCESS;
```

File: drivers/platform/mbed/mbed_i2c.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	i2c = (I2C *)(((mbed_i2c_desc *)(desc->extra))->i2c_port);
+
+	/**
+	    The MBED I2C API is reversed for parameter 4
+	    Instead of stop_bit - it has
+	    @param repeated   - Repeated start, true - don't send stop at end default value is false.
+	    Inverting here to keep the no-OS/platform_drivers API
+	 */
+	if (!(i2c->read(desc->slave_address, (char *)data, bytes_number, !stop_bit))) {
+		return SUCCESS;
```
Comment (amiclaus) [inline drivers/platform/mbed/mbed_i2c.cpp:197]: ditto

**Review by amiclaus — COMMENTED** [review]
Also please provide commit description, not only the title

File: drivers/platform/mbed/mbed_i2c.cpp (hunk 1 above)
Comment (mphalke) [reply drivers/platform/mbed/mbed_i2c.cpp]: I have updated this code as per review comment.

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**